app/controllers/carts_controller.rb

```diff
@@ -1,20 +1,36 @@
 class CartsController < ApplicationController
 
+  before_action :find_cart
+
   def add
     dish = Dish.find(params[:id])
-    quantity = params[:amount].nil? ? 1 : params[:amount]
-    if session[:cart_id]
-      show
+    if params[:amount].to_i
```

Comment:
Don't think you need to call to_i on this line. You are just checking if there is a value in this key, right?

Reply:
No I want to catch if there is anything other than integer, it sends an error message if it isn't